src/main/java/Main.java

```diff
@@ -0,0 +1,12 @@
+import fruit.shop.FruitShopServeImpl;
+import fruit.shop.ShopServe;
+import java.io.IOException;
+
+public class Main {
+    public static void main(String[] args) throws IOException {
+        String fileFrom = "src/main/resources/test1";
+        String fileTo = "src/main/resources/test2";
+        ShopServe fruitShop = new FruitShopServeImpl();
+        fruitShop.serveShop(fileFrom, fileTo);

```

Comment:
```suggestion
        String fileFrom = "src/main/resources/test1";
        String fileTo = "src/main/resources/test2";
        List<String> records = fileReader.read(sourceFile);
        List<FruitTransaction> transactions = transactionParser.parse(records); 
        
        Map<Operation, OperationHandler> handlersMap = Map.of(
             Operation.BALANCE, new OperationHandler()
             ....
        )
        FruitTransactionStrategy strategy = new FruitTransactionStrategyImpl(handlersMap);
        
        FruitTransactionProcessor processor = new FruitTransactionProcessorImpl(strategy);
        processor.process(transactions);
        
        ReportCreator reportCreator = new CsvReportCreatorImpl();
        
        String report = reportCreator.createReport(FruitStorage.getFruitMap()));
        
        FileWriter writer = new FileWriterImpl();
        writer.write(destinationFile, report);
```

lets divide the logic into separate independent services, call it step by step

Reply:
It's implemented under the hood in FruitShopServeImpl.